### PR TITLE
SF-1906 Separate migration into distinct subprocess

### PR DESCRIPTION
**Pull request by Nateowami**
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/1735)

### CODE REVIEW

File: src/SIL.XForge.Scripture/Migrator.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+using System.IO;
+
+namespace SIL.XForge.Scripture;
+
+public class Migrator
+{
+    public static void RunMigrations(string environment)
+    {
+        // Spawn a sub-process to run the migrator
+        var process = new Process();
```
[the remark this answers is not in the document]
Comment (pmachapman) [reply src/SIL.XForge.Scripture/Migrator.cs:12]: You could fix this by changing this line to:
```cs
using var process = new Process();
```

File: src/SIL.XForge.Scripture/Migrator.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
+using System;
+using System.Diagnostics;
+using System.IO;
+
+namespace SIL.XForge.Scripture;
+
+public class Migrator
```
Comment (pmachapman) [inline src/SIL.XForge.Scripture/Migrator.cs:7]: This class could be made static.

**Review by pmachapman — CHANGES_REQUESTED** [review]
I ran this on my 370K note thread collection, and it ran great, taking well over one minute (the Jering timeout).

Thanks! This is a good, integrated solution.

**Review by Nateowami — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/1735)* status: 0 of 7 files reviewed, 3 unresolved discussions (waiting on @github-code-scanning[bot] and @pmachapman)

___
*[`src/SIL.XForge.Scripture/Migrator.cs` line 7 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/1735#-NQ1pCx56ba--9hMP0q7-r1-7:-NQ3M4KV4jkqqSgoIV2a:b-i21bbo) ([raw file](https://github.com/sillsdev/web-xforge/blob/8bf71fa55d448a839a15bab3b6466960631673c6/src/SIL.XForge.Scripture/Migrator.cs#L7)):*
<details><summary><i>Previously, pmachapman (Peter Chapman) wrote…</i></summary><blockquote>

This class could be made static.
</blockquote></details>

Thanks. Done.
___
*[`src/SIL.XForge.Scripture/Migrator.cs` line 11 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/1735#-NQ3MJgu4rPHD5AoCIIF:-NQ3MJgu4rPHD5AoCIIG:b-9w5ok) ([raw file](https://github.com/sillsdev/web-xforge/blob/8bf71fa55d448a839a15bab3b6466960631673c6/src/SIL.XForge.Scripture/Migrator.cs#L11)):*
> ```Smalltalk
>     public static void RunMigrations(string environment)
>     {
>         // Spawn a sub-process to run the migrator
> ```

I don't think this comment is particularly helpful.

I'm pretty sure I added it as an, um, prompt for GitHub Copilot so it would show me the standard way of spawning a subprocess.
___
*[`src/SIL.XForge.Scripture/Migrator.cs` line 12 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/1735#-NQ1pCx56ba--9hMP0q7-r1-12:-NQ3M5Ij-pmZjjEIsslJ:b-896fix) ([raw file](https://github.com/sillsdev/web-xforge/blob/8bf71fa55d448a839a15bab3b6466960631673c6/src/SIL.XForge.Scripture/Migrator.cs#L12)):*
<details><summary><i>Previously, pmachapman (Peter Chapman) wrote…</i></summary><blockquote>

You could fix this by changing this line to:
```cs
using var process = new Process();
```
</blockquote></details>

Done.

**Review by pmachapman — APPROVED** [review]

**Pull request merged**